docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,457 @@
+    uint32_t qe_id_issuer_chain_size;       ///< Size of qe_id_issuer_chain
+
+    uint8_t* creation_datetime;             ///< Time the endorsements were generated,
+                                            ///< null-terminated string
+    uint32_t create_datetime_size;          ///< The size of creation_datetime.
+
+} oe_sgx_endorsements_t;
+```
+
+### New <B>Public</B> Attestation functions
+
+These functions supersede the existing functions:
+1. `oe_get_evidence()` supersedes `oe_verify_report()`
+2. `oe_verify_evidence()` supersedes `oe_verify_report()`
+
+These functions will use plug-in attestation framework.  For more information please
+see the design document `CustomAttestation.md`.  In short, the actual implementation
+of these functions will depend on which plug-in is registered.  By default there will
+be a built-in SGX plug-in.
+
+For more information on the parameters `custom_claims`, `claims` or `opt_params`
+please see design document `CustomAttestation.md`.
+
+`common/sgx/attestation.c`
+```C
+/**
+ * Get evidence signed by the enclave platform along with the corresponding
+ * endorsements for use in attestation.
+ *
+ * This function returns the evidence and endorsements used in **local** or
+ * **remote** attestation.
+ *
+ * This function can only be called from a TEE/enclave.
+ *
+ * @param[in] flags Specifying default value (0) generates evidence for local
+ * attestation. Specifying OE_EVIDENCE_FLAGS_REMOTE_ATTESTATION generates
+ * evidence for remote attestation.
+ * @param[in] custom_claims A buffer to the optional custom claims.
+ * @param[in] custom_claims_size The size in bytes of custom_claims.
+ * @param[in] opt_params Optional additional parameters needed for the current
+ * enclave type.
+ *     For SGX:
+ *        This can be sgx_target_info_t for local attestation.
+ * @param[in] opt_params_size The size of the **opt_params** buffer.
+ * @param[out] evidence_buffer This points to the resulting evidence upon success.
+ * @param[out] evidence_buffer_size This is set to the size of the evidence buffer
+ * on success.
+ * @param[out] endorsements_buffer The buffer containing the endorsements to parse.
+ * @param[out] endorsements_buffer_size The size of the **endorsements_buffer**.
+ *
+ * @retval OE_OK The evidence and endorsements were successfully created.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval OE_OUT_OF_MEMORY Failed to allocate memory.
+ *
+ */
+oe_result_t oe_get_evidence(
+    uint32_t flags,
+    const uint8_t* custom_claims,
+    size_t custom_claims_size,
+    const void* opt_params,
+    size_t opt_params_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size);
+
+/**
+ * Free up any resources allocated by oe_get_evidence()
+ *
+ * @param[in] evidence_buffer THe buffer containing the evidence.
```

Comment:
>H [](start = 31, length = 1)

minor - capitalization

Reply:
The `RemoteAttestationCollaterals.md` design doc is already merged in, but Ill fix that typo in a future PR.